src/deploy/functions/runtimes/node/index.ts

```diff
@@ -208,48 +195,81 @@ export class Delegate {
       const binPath = path.join(nodeModulesPath, ".bin", "firebase-functions");
       if (fileExistsSync(binPath)) {
         logger.debug(`Found firebase-functions binary at '${binPath}'`);
-        // Note: We cannot use inherit because we need the stdout/err to be
-        // omitted in commands that use --json.
-        const childProcess = spawn(binPath, [this.sourceDir], {
-          env,
-          cwd: this.sourceDir,
-          stdio: [/* stdin=*/ "ignore", /* stdout=*/ "pipe", /* stderr=*/ "pipe"],
-        });
-        childProcess.stdout?.on("data", (chunk: Buffer) => {
-          logger.info(chunk.toString("utf8"));
-        });
-        childProcess.stderr?.on("data", (chunk: Buffer) => {
-          logger.error(chunk.toString("utf8"));
-        });
-        return Promise.resolve(async () => {
-          const p = new Promise<void>((resolve, reject) => {
-            childProcess.once("exit", resolve);
-            childProcess.once("error", reject);
-          });
-
-          try {
-            await fetch(`http://localhost:${port}/__/quitquitquit`);
-          } catch (e) {
-            logger.debug(
-              "Failed to call quitquitquit. This often means the server failed to start",
-              e,
-            );
-          }
-          setTimeout(() => {
-            if (!childProcess.killed) {
-              childProcess.kill("SIGKILL");
-            }
-          }, 10_000);
-          return p;
-        });
+        return binPath;
       }
     }
+
     throw new FirebaseError(
       "Failed to find location of Firebase Functions SDK. " +
         "Please file a bug on Github (https://github.com/firebase/firebase-tools/).",
     );
   }
 
+  private spawnFunctionsProcess(
+    config: backend.RuntimeConfigValues,
+    envs: backend.EnvironmentVariables,
+  ): ChildProcess {
+    const env: NodeJS.ProcessEnv = {
+      ...envs,
+      FUNCTIONS_CONTROL_API: "true",
+      HOME: process.env.HOME,
+      PATH: process.env.PATH,
+      NODE_ENV: process.env.NODE_ENV,
+      // Web Frameworks fails without this environment variable
+      __FIREBASE_FRAMEWORKS_ENTRY__: process.env.__FIREBASE_FRAMEWORKS_ENTRY__,
+    };
+    if (Object.keys(config || {}).length) {
```

Comment:
Based on types, this || {} shouldn't be necessary. Either remove or fix the arg type?

Reply:
For the config || {} thing - might need it since config comes from runtime/API responses that could be null despite the types.
 
Added a comment explaining why we're being defensive here.